web-locks/non-fully-active.tentative.https.html

```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Web Locks API: Non-fully-active documents</title>
+<link rel=help href="https://wicg.github.io/web-locks/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/helpers.js"></script>
+
+<div></div>
+
+<script>
+  function createNonFullyActiveIframe(src) {
+    const iframe = document.createElement("iframe");
+    if (src) {
+      iframe.src = src;
+    }
+    document.body.appendChild(iframe);
+    const { navigator, DOMException, postMessage } = iframe.contentWindow;
+    iframe.remove();
+    return { iframe, navigator, DOMException, postMessage };
+  }
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.request("foo", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request() on non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.query();
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Query should explicitly fail");
+  }, "query() on a non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException, postMessage } = createNonFullyActiveIframe("resource/iframe.html");
+
+    const p = navigator.locks.request("-", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request()'s fully-active check happens earlier than name validation");
```

Comment:
I don't understand this comment.

Reply:
There is a step [to validate the name of the lock](https://wicg.github.io/web-locks/#dom-lockmanager-request):

>If name starts with U+002D HYPHEN-MINUS (-), then return a promise rejected with a "NotSupportedError" DOMException.

And this test checks throwing InvalidStateError happens first.

Reply:
BTW it's just for consistency with other specs, as mentioned in https://github.com/WICG/web-locks/pull/85:

>This follows Web Share behavior where fully-active check happens as early as possible.

Reply:
Aha, it is about request's name. I was wondering if it was referring to browsingcontext/window name, since this particular test passes explicit url. It isn't clear to me why the explicit url is used in this test.